### PR TITLE
Populate nshardwareplatform and nsosversion during join operation

### PR DESCRIPTION
**Pull request by Carbenium**
Fixes: https://pagure.io/freeipa/issue/8370

### CODE REVIEW

**Comment by rcritten** [comment]
These attributes need to be writeable for the enrollment role.

File: ipaserver/plugins/host.py [hunk 1]
```diff
@@ -361,7 +361,8 @@ class host(LDAPObject):
         },
         'System: Enroll a Host': {
             'ipapermright': {'write'},
-            'ipapermdefaultattr': {'objectclass', 'enrolledby'},
+            'ipapermdefaultattr': {'objectclass', 'enrolledby',
+                                   'nshardwareplatform', 'nsosversion'},
```
Comment (tiran) [inline ipaserver/plugins/host.py:365]:
```suggestion
            'ipapermdefaultattr': {
                'objectclass', 'enrolledby', 'nshardwareplatform', 'nsosversion'
            },
```

File: ipaserver/plugins/join.py [hunk 2]
```diff
@@ -85,7 +85,7 @@ class join(Command):
     has_output = tuple()
     use_output_validation = False
 
-    def execute(self, hostname, **kw):
+    def execute(self, hostname, **kwargs):
```
Comment (tiran) [inline ipaserver/plugins/join.py:88]: There is nothing wrong with ``kw`` here.

File: ipaserver/plugins/join.py [hunk 3]
```diff
@@ -94,13 +94,21 @@ def execute(self, hostname, **kw):
...
+        assert 'cn' not in kwargs
         ldap = self.api.Backend.ldap2
 
+        try:
+            # We don't use the realm parameter at the moment. Since we pass
+            # kwargs directly to host_{add,mod} and 'realm' is not a supported
+            # parameter, we remove it.
+            del kwargs['realm']
+        except KeyError:
+            pass
```
Comment (tiran) [inline ipaserver/plugins/join.py:106]:
```suggestion
        # realm parameter is not supported by host_{add,mod}
        kwargs.pop("realm", None)
```

Comment (Carbenium) [reply ipaserver/plugins/join.py]: I even considered this one :wink:

File: ipaserver/plugins/join.py (hunk 2 above)
Comment (Carbenium) [reply ipaserver/plugins/join.py]: True that.

**Comment by abbra** [comment]
LGTM

**Comment by abbra** [comment]
master:

* 8f640f8672f2bec7b560a2838e2451a190abf6cf Populate nshardwareplatform and nsosversion during join operation

**Pull request closed without merging**